clients/client-py/test/test_client.py

```diff
@@ -645,311 +645,307 @@ def fakeSite(url, request):
-            return httmock.response(200, content, headers, None, 5, request)
-
-        with httmock.HTTMock(auth_response):
-            client = subject.Auth({"rootUrl": "https://tc-tests.example.com", "credentials": {}})
-            result = client.client('abc')
-            self.assertEqual(result, {"clientId": "abc"})
-
-    def test_permacred_simple(self):
-        """we can call methods which require authentication with valid
-        permacreds"""
-        client = subject.Auth({
-            'rootUrl': self.real_root_url,
-            'credentials': {
-                'clientId': 'tester',
-                'accessToken': 'no-secret',
-            }
-        })
-        result = client.testAuthenticate({
-            'clientScopes': ['test:a'],
-            'requiredScopes': ['test:a'],
-        })
-        self.assertEqual(result, {'scopes': ['test:a'], 'clientId': 'tester'})
-
-    def test_permacred_simple_authorizedScopes(self):
-        client = subject.Auth({
-            'rootUrl': self.real_root_url,
-            'credentials': {
-                'clientId': 'tester',
-                'accessToken': 'no-secret',
-            },
-            'authorizedScopes': ['test:a', 'test:b'],
-        })
-        result = client.testAuthenticate({
-            'clientScopes': ['test:*'],
-            'requiredScopes': ['test:a'],
-        })
-        self.assertEqual(result, {'scopes': ['test:a', 'test:b'],
-                                  'clientId': 'tester'})
-
-    def test_unicode_permacred_simple(self):
-        """Unicode strings that encode to ASCII in credentials do not cause issues"""
-        client = subject.Auth({
-            'rootUrl': self.real_root_url,
-            'credentials': {
-                'clientId': u'tester',
-                'accessToken': u'no-secret',
-            }
-        })
-        result = client.testAuthenticate({
-            'clientScopes': ['test:a'],
-            'requiredScopes': ['test:a'],
-        })
-        self.assertEqual(result, {'scopes': ['test:a'], 'clientId': 'tester'})
-
-    def test_invalid_unicode_permacred_simple(self):
-        """Unicode strings that do not encode to ASCII in credentials cause issues"""
-        with self.assertRaises(exc.TaskclusterAuthFailure):
-            subject.Auth({
-                'rootUrl': self.test_root_url,
-                'credentials': {
-                    'clientId': u"\U0001F4A9",
-                    'accessToken': u"\U0001F4A9",
-                }
-            })
-
-    def test_permacred_insufficient_scopes(self):
-        """A call with insufficient scopes results in an error"""
-        client = subject.Auth({
-            'rootUrl': self.real_root_url,
+        assert self.gotUrl == 'https://tc-tests.example.com/api/fake/v1/two_args_with_input/a/b'
+        assert json.loads(self.gotRequest.body) == {"x": 1}
+
+
+@pytest.mark.skipif(os.environ.get('NO_TESTS_OVER_WIRE'), reason = "Skipping tests over wire")
```

Comment:
Same here: this decorator needs to go before every test function. Alternately, you can skip every test in a file if you do something like

```
pytestmark = [
    pytest.mark.skipif(os.environ.get("NO_TESTS_OVER_WIRE"), reason="Skipping tests over wire")
]
```

at the top of the file after the imports.

Reply:
@escapewindow Got it, Thanks!
I'll update thr PR.